plugin.video.adultswim/resources/lib/metahandler.py

```diff
@@ -0,0 +1,450 @@
+# -*- coding: utf-8 -*-
+"""
+Credit to script.module.metahndler
```

Comment:
I removed the MetaHandler (script.module.metahandler) dependency because it no longer seems to be available through the Official Kodi Repository. So upon installation, the dependency could not be imported and the addon would fail to install.

The version that was previously available (2.9.1) from the Kodi repo was broken anyway. I have updated and fixed this module which is available from 3rd party repositories but it requires a working TMDB api key which TMDB seems to not want to allow anymore since they ban newly created metahandler keys.

So the result is me stripping down the metahandler to include only what I needed to fetch the metadata from TVDB for the tv shows available. This was much easier and faster instead creating something from scratch.

Reply:
Metahandler is still in Kodis repository: https://github.com/xbmc/repo-scripts/tree/gotham/script.module.metahandler

As for it being broken, developers can still supply their own.
 `from metahandler import metahandlers
    metaget = metahandlers.MetaData(preparezip=False, tmdb_api_key=TMDB_API_KEY)`

Reply:
Yes I know how to use my own key (see here https://github.com/xbmc/repo-plugins/pull/1672/commits/111773d66e5eb1812571d5e3ed35810fa1cb160f#diff-f48585cc17636a91377ff9ed608fcc74L17). I see why it was failing to import. I was importing 2.9.1 instead of the available version 2.9.0 but still I rather use my "lite" version for the metadata.

There are a few other things broken in the 2.9.0 version...

Reply:
@eldorados needs to create a PR with his updated module otherwise use v.2.9.0.

Reply:
I'm fine if you want to use your light version

Reply:
@MartijnKaijser Yeah let's go with the light version for now since that's all I need.

@Lunatixz Eldorado hasn't developed the meta handler in quite sometime. He also recently deleted his repos but they are up again. I have been making the changes in the meta handler with him overseeing it but he is busy. I will ask him if he is willing to submit the updates or I can do it for him. While we are on this subject, would there be an issue using the tmdb key from the common.tmdb package for use in the meta handler?

Reply:
using that key is ABSOLUTELY FORBIDDEN in ANY WAY!
Just to be clear :)

Reply:
I understand. Which is why I asked.

Reply:
I didn't agree with breaking down modules for individual use (unless modified beyond the scope of the original module). Updating the standalone module should have a higher priority.

Reply:
@Lunatixz I have already updated the metahandler (see here https://github.com/jsergio123/script.module.metahandler). The original git was lost when Eldorado deleted his github. He then returned and forked my metahandler repo but most of the changes are lost so you wont be able to view the diffs. I would be willing to submit the update if you're willing to not scrutinize it.

@MartijnKaijser is ok with me using the lite version so i really prefer to use it so i dont have to deal with tmdb and them banning my api key like they always do. Plus i have no need for the majority of the logic in the metahandler for this addon. Its just unecessary code that results in unnecessary requests to additional websites and apis.

Reply:
@jsergio123 I understand, why don't you push the metahandler update yourself. I'd advise removing default keys and documenting the need for individual developer keys.

I don't see the logic of pushing updates that only benefit a single developer. 

Modules benefit all developers, if you have a update and the original author has abandoned the project it's kinda in your court to update the module. 

Where would we be if every time a module breaks only individual developers possess the solution.

Don't get me wrong if you didn't already fix the module and only possessed this "lite" solution no one would force you to update the module. However in this case the works already done, so I don't see the logic in compromise.

@MartijnKaijser may disagree, however being a active plug-in developer I do not see the reasoning behind keeping module updates to yourself :)

Reply:
@Lunatixz Ok I will send over the PR but the updates were always available via 3rd party repositories which I won't name ;)

I have asked Eldorado on numerous occasions to try and push the updates to the org repo which he suggested you guys would probably just laugh at us lol.

Reply:
I don't see any reason to laugh at @Eldorados he has a long history of supporting the official community with his modules and to my knowledge has a neutral stance elsewhere. I'm not familiar with you, and a google search didn't help... However if that's all in the past... so be it.

I look forward to seeing the metahandler update as i'm sure many developers do as well. Thanks

Reply:
URL resolver is a tool, unfortunately its been abused and perverted by pirates. I believe the teams only issue is the file servers/hosts. If the project went back on track and resolved various links akind to youtubedl you wouldn't find a issue from me.

Reply:
I reviewed your metahandler commit, the update did not affect functionality and because I rarely commit plugins that include "lite modules" I recommend you use the currently available module or remove the metahandler functionality all together. Thanks

Reply:
No one is trolling you; I was concerned that you included module components rather then using the module itself. If I overlooked something with the metahandler we can get that corrected, however you should understand sometimes submissions require further review. 

In this case there are a few hangups with metahandler; otherwise the merger would have went through. Why is there a issue with the "lite" code... because it's redundant and bad practice. You shouldn't take the review process personally.

Reply:
I'm not trying to go back and forth with you but since when is reducing file size, removing unnecessary code and http requests to 4 separate websites when only one is needed, removing a dependency, and increasing speed and reliability "redundant and bad practice"?

Reply:
@jsergio123 Well it's bad practice in our ecosystem, as we want everybody to help each other out and share as much code as possible. Another reason is that it will speed up our reviews, as we would need to review every PR that comes with a lite module carefully. On the other hand, if we have a module reviewed carefully and you just include it via `addon.xml`, it will be easier to review for us as we're already trusting the module.

> I'm not trying to go back and forth with you but since when is reducing file size, removing unnecessary code and http requests to 4 separate websites when only one is needed, removing a dependency, and increasing speed and reliability "redundant and bad practice"?

This sounds more like the module doesn't fit your specifications in the first place, so maybe using it is not the right solution and there has been a missunderstanding.

Reply:
@Razzeee but I'm not using the module and I don't see why @Lunatixz is forcing me to use the broken module.

Reply:
Well his previous post here explains it pretty well I think

> I didn't agree with breaking down modules for individual use (unless modified beyond the scope of the original module). Updating the standalone module should have a higher priority.

I don't think he ment to force you, just to guide you. We all benefit from better modules. If your lite module is fine, it's fine. But as a reviewer I hate such lite modules and it raises questions when reviewing and you find code credits to other modules.
So I would also prefere you to use a module, as long as it has the functionallity for you. And I'm pretty sure he doesn't want anybody to use a broken module :D
Probably just a communication problem between what qualifies as "broken" or different usecases. If it's only the show year fix, I would not exactly say it's broken, that's probably at most 10% of calls?